### PR TITLE
LibJS: Refactor Array.prototype callback functions and make them generic

### PR DESCRIPTION
**Pull request by linusg**
There are more to be made generic, but doing the ones taking a callback and optional this value was most straightforward as they have more or less the same logic every time.

### CODE REVIEW

File: Libraries/LibJS/Tests/Array.prototype-generic-functions.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+load("test-common.js");
+
+try {
+    const o = { length: 5, 0: "foo", 1: "bar", 3: "baz" };
+
+    ["every"].forEach(name => {
```
Comment (Dexesttp) [inline Libraries/LibJS/Tests/Array.prototype-generic-functions.js:6]: I'is there a reason for "every" to be away from "filter", forEach", etc..., or is it leftover from debugging ?

Comment (linusg) [reply Libraries/LibJS/Tests/Array.prototype-generic-functions.js:6]: It's on purpose, the latter have `return false` in the callback and `every` has `return true`, so it doesn't abort after the first item and visits all of them for the sake of testing.

Comment (Dexesttp) [reply Libraries/LibJS/Tests/Array.prototype-generic-functions.js:6]: Oh, didn't see that !

**Comment by awesomekling** [comment]
These are great!

**Pull request merged**